server/internal/graphql/resolver/configuration.go

```diff
@@ -22,6 +22,7 @@ func (r *Resolver) subscribeConfiguration(ctx context.Context, provider models.P
 	chp <- struct{}{}
 	cha <- struct{}{}
 	chf <- struct{}{}
+	key := len(r.Config.ConfigurationChannel.ApplicationsChannel)
```

Comment:
@MUzairS15 Given the fact that multiple instances of this function will be called for each client, here this key will introduce another race condition. 
Imagine this: Two clients connect at approx the same time. In both instances of this function, len will turn out to be the same and hence two different subscription will get the same key.
Use UUID or something like that.